src/Microsoft.OData.Core/JsonLight/ODataJsonLightServiceDocumentSerializer.cs

```diff
@@ -132,5 +133,131 @@ private void WriteServiceDocumentElement(ODataServiceDocumentElement serviceDocu
             // "}"
             this.JsonWriter.EndObjectScope();
         }
+
+        /// <summary>
+        /// Asynchronously writes a service document in JsonLight format.
+        /// </summary>
+        /// <param name="serviceDocument">The service document to write.</param>
+        internal Task WriteServiceDocumentAsync(ODataServiceDocument serviceDocument)
+        {
+            Debug.Assert(serviceDocument != null, "serviceDocument != null");
+
+            return this.WriteTopLevelPayloadAsync(
+                async () =>
+                {
+                    // "{"
+                    await this.AsynchronousJsonWriter.StartObjectScopeAsync()

```

Comment:
Is it too late to rename this to `AsyncJsonWriter` :)? Or the use of `Asynchronous` is standard across the stack?

Reply:
@habbes I looked through the repo and `Async` was mostly used a postfix while prefix situations mostly used `Asynchronous` or `asynchronous`. There are exceptions of course like in `AsyncBufferedStream`.
My eyes and brain already got used to `Asynchronous` but if people felt strongly for `Async` I could still go ahead and rename from any of the open pull requests

Reply:
I am biased towards use of `Async` instead of `Asynchronous`.
Built-in c# classes use `Async` e.g `IAsyncDisposable`, `IAsyncEnumerable`

Reply:
@KenitoInc Noted. Thanks